File: graphql-todo-items-java/src/main/java/com/example/subgraphtemplatejavadgsshows/todoitems/mutations/TodoItemsMutations.java
package com.example.subgraphtemplatejavadgsshows.todoitems.mutations;

import com.example.subgraphtemplatejavadgsshows.todoitems.TodoItemsService;
import com.netflix.graphql.dgs.DgsComponent;
import com.netflix.graphql.dgs.DgsMutation;
import org.openapitools.client.models.ToDoItem;

import java.io.IOException;

@DgsComponent
public class TodoItemsMutations {
    private final TodoItemsService itemsService;

    public TodoItemsMutations(TodoItemsService itemsService) {
        this.itemsService = itemsService;
    }

    @DgsMutation
    public String RemoveItem(String itemId) throws IOException {
        itemsService.removeItem(itemId);
        return "Item removed successfully";
    }

    @DgsMutation
    public ToDoItem AddTodo(String listId, String text) throws IOException {
        return itemsService.addTodo(listId, text);
    }

    @DgsMutation
    public ToDoItem ToggleTodoCheck(String itemId, boolean checked) throws IOException {
        return itemsService.toggleTodoCheck(itemId, checked);
    }

    @DgsMutation
    public ToDoItem RenameTodo(String todoId, String newText) throws IOException {
        return itemsService.renameTodo(todoId, newText);
    }
}

File: graphql-todo-items-java/src/main/java/com/example/subgraphtemplatejavadgsshows/todoitems/mutations/TodoItemsMutations.kt
package com.example.subgraphtemplatejavadgsshows.todoitems

import com.netflix.graphql.dgs.DgsComponent
import com.netflix.graphql.dgs.DgsMutation
import org.openapitools.client.apis.TodoItemControllerApi
import org.openapitools.client.models.ItemResource
import org.openapitools.client.models.RenameResource
import org.openapitools.client.models.ToDoItem
import java.util.UUID

@DgsComponent
class TodoItemsMutations {
    private val api = TodoItemControllerApi()

    @DgsMutation
    fun RemoveItem(itemId: String): String {
        api.delete(UUID.fromString(itemId))
        return "Item removed successfully"
    }

    @DgsMutation
    fun AddTodo(listId: String, text: String): ToDoItem {
        val itemResource = ItemResource(text, UUID.fromString(listId))
        return api.createItem(itemResource)
    }

    @DgsMutation
    fun ToggleTodoCheck(itemId: String, checked: Boolean): ToDoItem {
        return if (checked) api.markChecked(UUID.fromString(itemId)) else api.markUnchecked(UUID.fromString(itemId))
    }

    @DgsMutation
    fun RenameTodo(todoId: String, newText: String): ToDoItem {
        val renameResource = RenameResource(newText, UUID.fromString(todoId))
        return api.rename(renameResource)
    }
}
